results-processor/wptreport.py

```diff
@@ -244,8 +290,9 @@ def product_id(self, separator='-', sanitize=False):
         # os_version isn't required.
         if self.run_info.get('os_version'):
             name += separator + self.run_info['os_version']
-        assert len(self.hashsum) > 0, 'Missing hashsum of the report'
-        name += separator + self.hashsum[:10]
+        hashsum = self.hashsum()
+        assert len(hashsum) > 0, 'Missing hashsum of the report'
```

Comment:
Since this is an assert, I guess there's no input which can cause this assert to fail? (Seems like that's the distinction being made elsewhere, which I agree with.)

Reply:
Yes! It could only happen if I misused the hash library or something went wrong inside.